src/www/ui/core-auth.php

```diff
@@ -252,6 +260,31 @@ function OutputOpen()
    */
   function checkUsernameAndPassword($userName, $password)
   {
+    /* Check the user for external authentication */
+    if ($this->authExternal['useAuthExternal']) {
+      $username = $this->authExternal['loginAuthExternal'];
+      /* checking if user exist */
+      try {
+        $this->userDao->getUserAndDefaultGroupByUserName($username);
+      } catch (Exception $e) {
+        /* If user does not exist then we create it */
+        $User = trim(str_replace("'", "''", $this->authExternal['loginAuthExternal']));
+        $Pass = $this->authExternal['passwordAuthExternal'] ;
+        $Seed = rand() . rand();
+        $Hash = sha1($Seed . $Pass);
+        $Desc = $this->authExternal['descriptionAuthExternal'];
+        $Perm = 3;
+        $Folder = 1;
+        $Email_notify = "y";
+        $Email = $this->authExternal['emailAuthExternal'];
+        /* Set default list of agents when a new user is created */
+        $agentList = "agent_bucket,agent_copyright,agent_keyword,agent_mimetype,agent_monk,agent_nomos,agent_ojo";
+        $default_bucketpool_fk = 2;
+        $this->user-add_user($User,$Desc,$Seed,$Hash,$Perm,$Email,
```

Comment:
You mean? https://github.com/fossology/fossology/blob/8a17b9532f3b729451a126481555a59af5facaa5/src/lib/php/common-users.php#L44